python-checks/src/main/java/org/sonar/python/checks/hotspots/GraphQLIntrospectionCheck.java

```diff
@@ -61,31 +61,32 @@ private static void checkGraphQLIntrospection(SubscriptionContext ctx) {
     Optional.ofNullable(callExpression.calleeSymbol())
       .map(Symbol::fullyQualifiedName)
       .filter(GRAPHQL_VIEWS_FQNS::contains)
-      .filter(fqn -> hasSensitiveMiddlewares(callExpression.arguments()) ||
-        hasSensitiveValidationRules(callExpression.arguments()))
-      .ifPresent(fqn -> ctx.addIssue(callExpression, MESSAGE));
+      .filter(fqn -> !hasSafeMiddlewares(callExpression.arguments()))
+      .filter(fqn -> !hasSafeValidationRules(callExpression.arguments()))
+      .ifPresent(fqn -> ctx.addIssue(callExpression.callee(), MESSAGE));
   }
 
-  private static boolean hasSensitiveMiddlewares(List<Argument> arguments) {
+  private static boolean hasSafeMiddlewares(List<Argument> arguments) {
     RegularArgument argument = TreeUtils.argumentByKeyword("middleware", arguments);
     if (argument == null) {
-      return true;
+      return false;
     }
+
     return extractArgumentValues(argument)
-      .map(values -> values.isEmpty() || expressionsNameContainIntrospection(values))
-      .orElse(false);
+      .map(values -> !values.isEmpty() && expressionsNameContainIntrospection(values))
```

Comment:
Here we check for expression names (and I think it's good), but the ticket mentions class and method names.
So I guess something like:

```

    class IntrospectionCustomMiddleware:
        ...
    my_custom_middleware = IntrospectionCustomMiddleware()

    GraphQLView.as_view(
            name="introspection",
            schema=schema,
            graphiql=True,
            middleware=[my_custom_middleware]
        )
```
should not raise an issue, but it does.

(Same thing applies to validation rules)
Sorry this was missed in the first review...

Reply:
Yes, no worries do you think it is enough to check the runtime type for this one? I will do an implementation for this I guess some other cases would then miss and I will create a ticket. i.e. for the case where a method returns safe middleware.

Reply:
I think checking runtime time is enough indeed!
The case where a method not named `introspection` returns a safe middleware would probably be too complex for a rule like this one, I don't think we need to cover it.